lib/Installation/Partitioner/ExpertPartitionerPage.pm

```diff
@@ -109,4 +115,30 @@ sub press_accept_button {
     send_key('alt-a');
 }
 
+sub open_clone_partition_dialog {
+    my ($self) = @_;
+    assert_screen(EXPERT_PARTITIONER_PAGE);
+    send_key($self->{partition_table_shortcut});
+    for (1 .. 2) { wait_screen_change { send_key "down" } }
+    send_key "ret";
+}
+
+sub select_all_disks_to_clone {
+    my ($self) = @_;
+    assert_screen(CLONE_PARTITION);
+    send_key($self->{available_target_disks_shortcurt});
+    until (check_screen(LAST_DISK_SELECTED)) {
```

Comment:
AFAIK send_key_until_needlematch only passes one command, not two (?) anyway it would also cause the infinite loop I was mentioning in the chat if the needle does not match. I found another solution that I will push shortly.

Reply:
``` perl
send_key_until_needlematch <needle_to_match> "spc"; 
wait_screen_change { send_key "down" };
```
what about that?
Also check line 131, you have forgotten a ';'

Reply:
Still, infinite loop.